File: my-folder/0078-subsets/solution.java
class Solution {
    public List<List<Integer>> subsets(int[] nums) {
        List<List<Integer>> res = new ArrayList<>();
        List<Integer> list = new ArrayList<>();
        helper(res, list, 0, nums);
        return res;
    }
    
    private void helper(List<List<Integer>> res, List<Integer> list, int start, int[] nums) {
        res.add(new ArrayList<Integer>(list));
        
        for (int i = start; i < nums.length; i++) {
            list.add(nums[i]);
            helper(res, list, i+1, nums);
            list.remove(list.size() - 1);
        }
    }
}

File: my-folder/0078-subsets/solution.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        self.res = []
        path = []
        self.backtrack(nums, 0, path)
        return self.res

    def backtrack(self, nums, start, path):
        self.res.append(path[:])
        for i in range(start, len(nums)):
            path.append(nums[i])
            self.backtrack(nums, i+1, path)
            path.pop()
    

